src/EnergyPlus/UnitarySystem.cc

```diff
@@ -15446,6 +15641,80 @@ namespace UnitarySystems {
         return Residuum;
     }
 
+    Real64 UnitarySys::calcMultiStageSuppCoilLoadResidual(EnergyPlusData &state,
+                                                          Real64 const SpeedRatio,
+                                                          std::vector<Real64> const &Par // Function parameters
+    )
+    {
+        // FUNCTION INFORMATION:
+        //       AUTHOR         Yueyue Zhou
+        //       DATE WRITTEN   May 2022
+
+        // PURPOSE OF THIS FUNCTION:
+        // Calculates part load ratio for multistage supplemental coil.
+
+        // METHODOLOGY EMPLOYED:
+        // Calls calc routines of  multistage electric Coil to get delivered heating at the given stage
+        // and calculates the residual as defined above
+
+        // Return value
+        Real64 Residuum; // residual to be minimized to zero
+
+        // FUNCTION LOCAL VARIABLE DECLARATIONS:
+        Real64 CycRatio;
+        Real64 QActual;
+        int SpeedNum;
+        int FanOpMode;
+
+        int CoilIndex = int(Par[1]);
+        CycRatio = Par[4];
+        SpeedNum = int(Par[5]);
+        FanOpMode = int(Par[6]);
+
+        HeatingCoils::CalcMultiStageElectricHeatingCoil(state, CoilIndex, SpeedRatio, CycRatio, SpeedNum, FanOpMode, QActual, true);
+
+        Residuum = Par[2] - QActual;
+
+        return Residuum;
+    }
```

Comment:
I mean technically, this function is 7 lines long:

```
Real64 UnitarySys::calcMultiStageSuppCoilLoadResidual(EnergyPlusData &state,
                                                          Real64 const SpeedRatio,
                                                          std::vector<Real64> const &Par // Function parameters
    )
    {
        int CoilIndex = int(Par[1]);
        Real64 CycRatio = Par[4];
        int SpeedNum = int(Par[5]);
        int FanOpMode = int(Par[6]);
        Real64 QActual;
        HeatingCoils::CalcMultiStageElectricHeatingCoil(state, CoilIndex, SpeedRatio, CycRatio, SpeedNum, FanOpMode, QActual, true);
        return Par[2] - QActual;
    }
```

I understand we may want a couple comments, but your authorship is available in Git history, we don't need it in comments, and you can declare variables where they are assigned, and you don't need an extra variable to store the return value just to return it.  I get that this just mimcs other functions from in here, but if other changes end up being needed, it would be nice to at least use best-practice when adding brand new functions.

Reply:
So you are saying it should look like this?

        Real64 residual = Par[2] - QActual;
        return residual;